src/test/java/FishingExample.java

```diff
@@ -0,0 +1,38 @@
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.projectile.FishingBobberEntity;
+import net.minecraft.util.math.EntityRayTraceResult;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraft.util.math.vector.Vector3d;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.event.entity.ProjectileImpactEvent;
+import net.minecraftforge.fml.common.Mod;
+
+@Mod("fishing_example")
+public class FishingExample
+{
+    public FishingExample()
+    {
+        MinecraftForge.EVENT_BUS.register(this);
+        MinecraftForge.EVENT_BUS.addListener(FishingExample::handleImpact);
+    }
```

Comment:
Please add an ENABLED boolean which if true runs everything in the constructor. See other test mods for an example.